### PR TITLE
MNT: `sparse` import refactor

### PR DESCRIPTION
**Pull request by mtsokol**
Hi @hameerabbasi,

This PR refactors `sparse` importing mechanism. It enforces backend selection at the import time and allows `from sparse import COO` etc. imports.

### CODE REVIEW

File: benchmarks/benchmark_backends.py [hunk 1]
```diff
@@ -1,27 +1,28 @@
...
     def setup(self):
+        skip_if_finch()
+
+        import sparse
+
         rng = np.random.default_rng(0)
 
         random_kwargs = {"density": 0.01, "random_state": rng}
-        if sparse.BackendType.Numba == BACKEND:
+        if sparse.BackendType.Numba == sparse.backend:
```
Comment (hameerabbasi) [inline benchmarks/benchmark_backends.py:19]: Can we remove the `if` here as we support just one backend here?

File: benchmarks/benchmark_backends.py [hunk 2]
```diff
@@ -1,27 +1,28 @@
...
             random_kwargs["format"] = "gcxs"
 
         self.s1 = sparse.random((100, 10), **random_kwargs)
         self.s2 = sparse.random((100, 100, 10), **random_kwargs)
 
-        if sparse.BackendType.Finch == BACKEND:
+        if sparse.BackendType.Finch == sparse.backend:
```
Comment (hameerabbasi) [inline benchmarks/benchmark_backends.py:25]: Same here

File: benchmarks/benchmark_backends.py [hunk 3]
```diff
@@ -46,18 +49,22 @@ class SpMv:
...
+        skip_if_finch()
+
+        import sparse
+
         rng = np.random.default_rng(0)
         size, density = size_and_density
 
         random_kwargs = {"density": density, "random_state": rng}
-        if sparse.BackendType.Numba == BACKEND:
+        if sparse.BackendType.Numba == sparse.backend:
```
Comment (hameerabbasi) [inline benchmarks/benchmark_backends.py:60]: Same here

File: benchmarks/benchmark_backends.py [hunk 4]
```diff
@@ -75,16 +82,13 @@ def fn(tns1, tns2, tns3):
                 def fn(tns1, tns2, tns3):
                     return tns1 @ tns2 + tns3
 
-        elif sparse.BackendType.Numba == BACKEND:
+        elif sparse.BackendType.Numba == sparse.backend:
```
Comment (hameerabbasi) [inline benchmarks/benchmark_backends.py:85]: Same here

File: benchmarks/benchmark_backends.py [hunk 5]
```diff
@@ -96,16 +100,20 @@ class Elemwise:
...
     def setup(self):
+        skip_if_finch()
+
+        import sparse
+
         rng = np.random.default_rng(0)
 
         random_kwargs = {"density": 0.01, "random_state": rng}
-        if sparse.BackendType.Numba == BACKEND:
+        if sparse.BackendType.Numba == sparse.backend:
```
Comment (hameerabbasi) [inline benchmarks/benchmark_backends.py:110]: Same here

File: benchmarks/benchmark_backends.py [hunk 6]
```diff
@@ -96,16 +100,20 @@ class Elemwise:
...
             random_kwargs["format"] = "gcxs"
 
         self.s1 = sparse.random((100, 10), **random_kwargs)
         self.s2 = sparse.random((100, 100, 10), **random_kwargs)
 
-        if sparse.BackendType.Finch == BACKEND:
+        if sparse.BackendType.Finch == sparse.backend:
```
Comment (hameerabbasi) [inline benchmarks/benchmark_backends.py:116]: Same here

File: benchmarks/utils.py [hunk 7]
```diff
@@ -4,5 +4,5 @@
 
 
 def skip_if_finch():
-    if sparse.backend_var.get() == sparse.BackendType.Finch:
+    if sparse.backend == sparse.BackendType.Finch:
```
Comment (hameerabbasi) [inline benchmarks/utils.py:7]: Same here.

File: benchmarks/benchmark_backends.py (hunk 1 above)
Comment (mtsokol) [reply benchmarks/benchmark_backends.py]: Done! Actually I removed whole `benchmark_backends.py` as Numba is benchmarked in other files and Finch will be run separately outside of asv.

File: benchmarks/benchmark_backends.py (hunk 2 above)
Comment (mtsokol) [reply benchmarks/benchmark_backends.py]: Done!

File: benchmarks/benchmark_backends.py [hunk 8]
```diff
@@ -46,18 +49,22 @@ class SpMv:
...
-        if sparse.BackendType.Numba == BACKEND:
+        if sparse.BackendType.Numba == sparse.backend:
             random_kwargs["format"] = "gcxs"
 
         self.M = sparse.random((size, size), **random_kwargs)
         self.v1 = rng.normal(size=(size, 1))
         self.v2 = rng.normal(size=(size, 1))
 
-        if sparse.BackendType.Finch == BACKEND:
+        if sparse.BackendType.Finch == sparse.backend:
```
[the remark this answers is not in the document]
Comment (mtsokol) [reply benchmarks/benchmark_backends.py]: Done!

File: benchmarks/benchmark_backends.py (hunk 4 above)
Comment (mtsokol) [reply benchmarks/benchmark_backends.py]: Done!

File: benchmarks/benchmark_backends.py (hunk 5 above)
Comment (mtsokol) [reply benchmarks/benchmark_backends.py]: Done!

File: benchmarks/benchmark_backends.py (hunk 6 above)
Comment (mtsokol) [reply benchmarks/benchmark_backends.py]: Done!

File: benchmarks/utils.py (hunk 7 above)
Comment (mtsokol) [reply benchmarks/utils.py]: Done!

**Comment by mtsokol** [comment]
> Maybe we can run the test suite twice, once for Finch and once for Numba in CI, without doing the `importlib.reload(...)` magic? And provide a `./test.sh` script?

Done! I run tests twice in the CI with different backends:
```yml
- name: Run tests
  run: |
    SPARSE_BACKEND=Numba pytest --pyargs sparse
    SPARSE_BACKEND=Finch pytest --pyargs sparse/tests
```
For Numba we run all tests. For Finch only the common ones.

**Review by hameerabbasi — APPROVED** [review]
This LGTM! Thanks so much for working on this, @mtsokol!

**Pull request merged**